### PR TITLE
convert kitchen tests to docker + other fixes

### PR DESCRIPTION
**Pull request by lamont-granquist**
- use kitchen-dokken instead of kitchen-ec2
- add tests for gem metadata + chef-sugar
- convert to inspec and add some tests

this disables the webapp stuff completely and its testing for now, that
needs to be updated to more modern httpd/mysql cookbooks and have its
testing converted to inspec.

more O/S distros are also in the pipeline.

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core want to get this merged to make some iterative progress, there's enough here that shipping it will be mostly a major level up.

**Comment by smurawski** [comment]
Let's do it! 👍

File: kitchen-tests/.kitchen.travis.yml [hunk 1]
```diff
@@ -20,22 +22,90 @@ provisioner:
...
+    intermediate_instructions:
+      - RUN /usr/bin/apt-get update
+      - RUN /usr/bin/apt-get -y install zlib1g-dev sudo net-tools
+
+- name: centos-5
+  driver:
+    image: centos:5
+    platform: rhel
+    run_command: /sbin/init
+    provision_command:
```
Comment (tas50) [inline kitchen-tests/.kitchen.travis.yml]: This should be intermediate_instructions

File: kitchen-tests/.kitchen.travis.yml [hunk 2]
```diff
@@ -20,22 +22,90 @@ provisioner:
...
+      - RUN yum clean all
+      - RUN yum install -y which initscripts net-tools sudo
+
+- name: centos-6
+  driver:
+    image: centos:6
+    intermediate_instructions:
```
Comment (tas50) [inline kitchen-tests/.kitchen.travis.yml]: Probably wanna add     run_command: /sbin/init

**Pull request merged**